app/Http/Controllers/Api/ApplicantController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\UpdateApplicantProfile;
+use App\Http\Resources\ApplicantProfile as ApplicantProfileResource;
+use App\Models\Applicant;
+use App\Models\ApplicantClassification;
+
+class ApplicantController extends Controller
+{
+
+    /**
+     * Retrieve Applicant profile.
+     *
+     * @param Applicant $applicant Incoming Applicant object.
+     *
+     * @return mixed
+     */
+    public function getProfile(Applicant $applicant)
+    {
+          $applicant->loadMissing('applicant_classifications');
+          return new ApplicantProfileResource($applicant);
+    }
+
+    /**
+     * Update Applicant profile.
+     *
+     * @param UpdateApplicantProfile $request Form Validation casted request object.
+     * @param Applicant              $applicant Incoming Applicant object.
+     */
+    public function updateProfile(UpdateApplicantProfile $request, Applicant $applicant)
+    {
+        $validatedRequest = $request->validated();
+        // If there are no applicant classifications in the request,
+        // then delete all applicant classifications attached to applicant.
+        if (!array_key_exists('applicant_classifications', $validatedRequest)) {
+            $applicant->applicant_classifications()->delete();
+        } else {
+            $newApplicantClassifications = collect($validatedRequest['applicant_classifications']);
+            $oldApplicantClassifications = $applicant->applicant_classifications;
+
+            // Delete old applicant classifications that were not resubmitted.
+            foreach ($oldApplicantClassifications as $applicantClassification) {
+                $newApplicantClassification = $newApplicantClassifications->firstWhere(
+                    'id',
+                    $applicantClassification['id']
+                );
+                if (!$newApplicantClassification) {
+                    $applicantClassification->delete();
+                }
+            }
+
+            // Update old applicant classifications and/or create them if it doesn't exist.
+            $newApplicantClassifications->map(function ($applicantClassification) use ($applicant) {
+                $newApplicantClassification = ApplicantClassification::firstOrNew(
+                    ['id' => $applicantClassification['id']]
+                );
```

Comment:
I don't think we can use firstOrNew with the 'id' field. In the case of a truly new entity, we want the created object to _ignore_ the id sent in the request, and use an auto-incremented id.

Reply:
I tested this out and logged the outputs, and it seems to be working as intended. If it's a new entity, it will use an auto-incremented id, and ignore the id sent in the request.